wolvic/browser/webdata_services/web_data_service_factory.h

```diff
@@ -0,0 +1,46 @@
+#ifndef WOLWIC_BROWSER_WEBDATA_SERVICES_WEB_DATA_SERVICE_FACTORY_H_
+#define WOLWIC_BROWSER_WEBDATA_SERVICES_WEB_DATA_SERVICE_FACTORY_H_
+
+#include "base/memory/ref_counted.h"
+#include "build/build_config.h"
+#include "components/keyed_service/core/service_access_type.h"
+#include "components/webdata_services/web_data_service_wrapper_factory.h"
+
+namespace base {
+template <typename T>
+class NoDestructor;
+}
+
+// Singleton that owns all WebDataServiceWrappers and associates them with
+// Profiles.
+class WebDataServiceFactory
+    : public webdata_services::WebDataServiceWrapperFactory {
+ public:
+  // Returns the WebDataServiceWrapper associated with the |profile|.
+  static WebDataServiceWrapper* GetForProfile(content::BrowserContext* profile,
```

Comment:
I guess `context` name seems more appropriate than `profile` name.

Reply:
yeah, I also changed some other method signatures to avoid mentioning the Profile concept.